### PR TITLE
Az.StackHCI Cmdlet registration changes for GA

### PR DESCRIPTION
**Pull request by dineshreddy007**
## Description

Design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/773

Made changes to registration script to register the GA version of On-Premises Azure Stack HCI with Azure.
- Supports registering with user provided certificate thumbprint.
- Supports On-Premises Azure Stack HCI OS changes to use independent certificate on cluster nodes.
- Cleans up resource group during unregistration.
- Improves registration output and logging.
Breaks the public preview registration of On-Premises Azure Stack HCI with Azure.
- To register public preview On-Premises Azure Stack HCi with Azure, use 0.3.1 version of Az.StackHCI.

## Checklist

- [Yes ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [Yes ] The title of the PR is clear and informative
- [Yes ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ No] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [Yes ] If applicable, the changes made in the PR have proper test coverage
- [Yes ] For public API changes to cmdlets:
    - [InProcess ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [Yes ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/StackHCI/help/Test-AzStackHCIConnection.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+external help file: Az.StackHCI-help.xml
+Module Name: Az.StackHCI
+online version:
+schema: 2.0.0
+---
+
+# Test-AzStackHCIConnection
+
+## SYNOPSIS
+
```
Comment (isra-fel) [inline src/StackHCI/help/Test-AzStackHCIConnection.md:11]: Please fill in synopsis

File: src/StackHCI/help/Test-AzStackHCIConnection.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+external help file: Az.StackHCI-help.xml
+Module Name: Az.StackHCI
+online version:
```
Comment (isra-fel) [inline src/StackHCI/help/Test-AzStackHCIConnection.md:4]: Use
```suggestion
online version: https://docs.microsoft.com/powershell/module/az.stackhci/test-azstackhciconnection
```

File: src/StackHCI/ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,13 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Made changes to registration script to register the GA version of On-Premises Azure Stack HCI with Azure.
+    - Supports registering with user provided certificate thumbprint.
+    - Supports On-Premises Azure Stack HCI OS changes to use independent certificate on cluster nodes.
+    - Cleans up resource group during unregistration.
+    - Improves registration output and logging.
+* Breaks the public preview registration of On-Premises Azure Stack HCI with Azure.
```
Comment (isra-fel) [inline src/StackHCI/ChangeLog.md:26]: More explicit

```suggestion
* [Breaking change] Breaks the public preview registration of On-Premises Azure Stack HCI with Azure.
```

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/StackHCI/help/Test-AzStackHCIConnection.md (hunk 1 above)
Comment (dineshreddy007) [reply src/StackHCI/help/Test-AzStackHCIConnection.md:12]: Filled in synopsis now.

File: src/StackHCI/help/Test-AzStackHCIConnection.md (hunk 2 above)
Comment (dineshreddy007) [reply src/StackHCI/help/Test-AzStackHCIConnection.md]: updated online version

File: src/StackHCI/ChangeLog.md (hunk 3 above)
Comment (dineshreddy007) [reply src/StackHCI/ChangeLog.md]: Mentioning explicitly now.

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**